TwitchLib.Client/TwitchClient.cs

```diff
@@ -1254,281 +1183,124 @@ private async Task HandleIrcMessageAsync(IrcMessage ircMessage)
+                MsgIds.RaidNoticeMature => Task.Run(() => OnRaidedChannelIsMatureAudience?.Invoke(this, null)), // TODO: Make event handler async
+                MsgIds.MsgBannedEmailAlias => OnBannedEmailAlias?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgChannelSuspended => HandleChannelSuspended(ircMessage),
+                MsgIds.MsgRequiresVerifiedPhoneNumber => OnRequiresVerifiedPhoneNumber?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgVerifiedEmail => OnRequiresVerifiedEmail?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.NoVIPs => OnVIPsReceived?.Invoke(this, new() { Channel = channel, VIPs = Array.Empty<string>() }),
+                MsgIds.VIPsSuccess => OnVIPsReceived?.Invoke(this, new()
+                {
+                    Channel = channel,
+                    // TODO: Make it less allocatey
+                    VIPs = message.AsSpan().SplitFirst(':').Remainder.ToString().Replace(" ", "").Replace(".", "").Split(',')
+                }),
+                MsgIds.MsgRateLimit => OnRateLimit?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgDuplicate => OnDuplicate?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgFollowersOnly => OnFollowersOnly?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgSubsOnly => OnSubsOnly?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgEmoteOnly => OnEmoteOnly?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgSuspended => OnSuspended?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgBanned => OnBanned?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgSlowMode => OnSlowMode?.Invoke(this, new() { Channel = channel, Message = message }),
+                MsgIds.MsgR9k => OnR9kMode?.Invoke(this, new() { Channel = channel, Message = message }),
+                _ => null
+            };
 
-                case MsgIds.MsgSlowMode:
-                    if (OnSlowMode != null)
-                        await OnSlowMode.Invoke(this,
-                            new OnSlowModeArgs
-                            {
-                                Channel = ircMessage.Channel,
-                                Message = ircMessage.Message
-                            });
-                    break;
+            return result ?? OnUnaccountedFor?.Invoke(this, new()
+            {
+                BotUsername = TwitchUsername,
+                Channel = channel,
+                Location = "NoticeHandling",
+                RawIRC = rawIrcMessage
+            }) ?? UnaccountedFor(rawIrcMessage);
```

Comment:
this part of the code is not good, we can get false positive `OnUnaccountedFor` just because `OnSlowMode?.Invoke(..)` return null
`UnaccountedFor` should by sync (someone forget change it back)

Reply:
`UnaccountedFor` returns `Task.CompletedTask` which is basically free.

As for `OnSlowMode` being null - do you mean that even if it doesn't have an event handler registered, we should it ignore and not raise `OnUnnacountedFor` or `UnaccountedFor`? E.g. changing `OnSlowMode` to null-coalesce to `Task.CompletedTask`? Are there other events for which this is a desired behavior?